Commands/Handlers/Member/ExtendMembership/ExtendMembershipHandler.cs

```diff
@@ -23,6 +23,11 @@ public async Task<Unit> Handle(ExtendMembershipCommand request, CancellationToke
         if (member == null)
             throw new ArgumentException($"No member found by Id {request.Id}", nameof(request.Id));
 
+        if (member.MembershipFee == 0)
```

Comment:
Why shouldn't it be allowed to extend the membership of a member who has membership fee of 0?

Reply:
Because it would make no sense? If the member has no membership fee, the membershipexpirationdate should not be set.

Reply:
Note that the membership expiration date also determines if a member is still active or not. Therefor I think we should still be able to set an expiration date on members who have a fee of 0. Otherwise there's no way to expire their membership

Reply:
Agreed, but we also should be able not to have a membership expiry date set. This allows us to have members that don't have to pay a membership fee.
So on a functional level members can:
- have no membership fee / a membership fee of 0 -> no membership expiry date set (active user)
- have a membership fee, and having paid their membership fee allowing the membership expiry date to be set in the future (active user)
- have a membership fee, and not having paid their membership fee. So their membership expiry date is in the past (inactive user)
Following these rules, it makes sense to have a check on the membership fee when extending a membership expiry date - but better would be to check if the membership expiry date is set (`!= null`)

Reply:
Agreed.
But since this is the `ExtendMembershipHandler`, handling the `ExtendMembershipCommand`, it should not refuse to handle the command based on the member's fee (which can be 0 or higher), this handler should not look at the fee amount and just carry out the command it was given, it's where the decision is made to publish this command that it should be checked if the fee is 0.

Reply:
I've removed the check.